ibis/backends/pandas/client.py

```diff
@@ -133,7 +168,19 @@ def infer_pandas_timestamp(value):
 
 @dt.infer.register(np.ndarray)
 def infer_array(value):
-    # TODO(kszucs): infer series
+    np_dtype_name = value.dtype.name
+    if np_dtype_name == 'object':
+        # The dtype of the np.array is ambiguous, so we need to infer the type
+        # of the elements manually.
+        if value.size == 0:
+            return dt.Array(dt.null)
```

Comment:
technically null is correct here but unfortunately "null" type is not supported in every system. 

I think it's probably OK to use sth like dt.Array(dt.int8) here

Reply:
> unfortunately "null" type is not supported in every system

@icexelloss do you mind clarifying this? Is this an issue with the Ibis `null` dtype in general?

Reply:
In the latest version this would now be inferred as `dt.Array(dt.binary)` (using `dt.binary` as the general fallback type)

Reply:
> do you mind clarifying this? Is this an issue with the Ibis null dtype in general?

yes, for example, the Spark arrow convertor don't support null type.